search/tests/test_es_indexing.py

```diff
@@ -255,3 +265,36 @@ def test_sorting(self):
             sec_res.id,
             res4.id
         )
+
+    def test_index_exception(self):
+        """
+        Ensure that an exception is thrown if we have not indexed properly.
+        """
+        # Delete the index
+        remove_index()
+
+        search_url = "/api/v1/repositories/{repo}/search/".format(
+            repo=self.repo.slug
+        )
+        with self.assertRaises(ReindexException):
+            self.client.get(search_url)
+
+        # Create a mapping
+        with self.assertRaises(ReindexException):
+            # Mapping doesn't exist yet
+            get_conn()
+
+        conn = get_conn(verify=False)
+        conn.indices.create(INDEX_NAME)
+        conn.indices.refresh()
+        create_mapping()
+
+        with self.assertRaises(Exception):
+            self.client.get(search_url)
```

Comment:
I'm getting a TransportError here with `No mapping found for [id] in order to sort on`. It only goes away when I index the resources. I'll investigate a fix for this